src/main/java/org/folio/rest/utils/CalendarUtils.java

```diff
@@ -306,10 +314,31 @@ public static OpeningPeriod mapOpeningsToOpeningPeriod(Openings openings) {
   }
 
   public static AsyncResult<Response> mapExceptionToResponseResult(Throwable e) {
+    Response errResponse;
     if (e.getClass() == NotFoundException.class) {
-      return succeededFuture(Response.status(404).header("Content-Type", "text/plain").entity(e.getMessage()).build());
+      errResponse = buildErrorResponse(404, e.getMessage());
+    } else if (e.getClass() == OverlapIntervalException.class) {
+      errResponse = buildErrorResponse(422, e.getMessage());
     } else {
-      return succeededFuture(Response.status(500).header("Content-Type", "text/plain").entity(e.getMessage()).build());
+      errResponse = buildErrorResponse(500, e.getMessage());
     }
+
+    return succeededFuture(errResponse);
+  }
+
+  private static Response buildErrorResponse(int status, String errMessage) {
+    return Response
+      .status(status)
+      .header(CONTENT_TYPE, TEXT_PLAIN)
```

Comment:
Please use "application/json" Content-Type for 422 error response

Reply:
Fix